### PR TITLE
Add option `Tables`

### PR DESCRIPTION
**Pull request by vs9h**
* Apply changes from `clang-format` in `src/test/` directory.
* Small code fixes (make non-static methods static, change naming, etc.)
* Add option `Tables` for multipath algorithms.

### CODE REVIEW

File: src/core/config/all_options.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ boost::program_options::options_description GeneralOptions() {
     po::options_description general_options("General options");
     general_options.add_options()
             (kCsvPath, po::value<std::filesystem::path>()->required(), kDCsvPath)
+            (kCsvPaths, po::value<std::vector<std::filesystem::path>>()->required(), kDCsvPaths)
```
Comment (BUYT-1) [inline src/core/config/all_options.cpp:45]: `->required` should be removed

File: src/core/config/tabular_data/input_tables/option.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// This option is meant for the case where an algorithm accepts several
+// tables.
```
Comment (BUYT-1) [inline src/core/config/tabular_data/input_tables/option.h:2]: I added the comment to input_table/option.h so it wouldn't accidentally be used if an algorithm uses 2 tables, for example. But I don't see how this option could be confusing, so this comment isn't needed.

**Review by BUYT-1 — CHANGES_REQUESTED** [review]

File: src/core/config/all_options.cpp (hunk 1 above)
Comment (vs9h) [reply src/core/config/all_options.cpp:45]: Also it seems, that we should remove `required()` for `kCsvPath`, isn't it?

Comment (vs9h) [reply src/core/config/all_options.cpp:45]: Also it seems, that we should remove `required()` for `kCsvPath`, isn't it?

File: src/core/config/tabular_data/input_tables/option.h (hunk 2 above)
Comment (vs9h) [reply src/core/config/tabular_data/input_tables/option.h:2]: Ack

File: src/core/config/all_options.cpp (hunk 1 above)
Comment (BUYT-1) [reply src/core/config/all_options.cpp]: yes, that's why I highlighted both lines

**Review by BUYT-1 — APPROVED** [review]
Looks okay

File: src/core/algorithms/algo_factory.h [hunk 3]
```diff
@@ -71,12 +72,28 @@ template <typename OptionMap>
...
         using namespace config::names;
+        namespace fs = std::filesystem;
         if (option_name == kTable && options.find(std::string{kTable}) == options.end()) {
             config::InputTable parser = std::make_shared<CSVParser>(
-                    details::ExtractOptionValue<std::filesystem::path>(options, kCsvPath),
+                    details::ExtractOptionValue<fs::path>(options, kCsvPath),
                     details::ExtractOptionValue<char>(options, kSeparator),
                     details::ExtractOptionValue<bool>(options, kHasHeader));
             return boost::any{parser};
+        } else if (option_name == kTables && options.find(std::string{kTables}) == options.end()) {
```
Comment (polyntsov) [inline src/core/algorithms/algo_factory.h:82]: This case handling looks bad, we certainly don't want here a big if-else spaghetti handling all cases where input data option may be constructed from several others. Let's leave it as it is for now, and I'll create an issue

**Review by polyntsov — APPROVED** [review]

**Pull request merged**